### ISSUE TITLE
New contributors guidelines.

### ISSUE THREAD

**Issue by gustavo-ren** [tensorflow/java#435]
Hello!

I am not sure if this is the right channel for this question but here we go, I am interested in starting contributing with open source projects, and Tensorflow seems to be a good option, I've read the CONTRIBUTING.md for the Java project, but I could not find guidelines for new contributors (such as take part on code reviews or write tests).

Can someone direct me on this matter?

Highly appreciate.

**Comment by Craigacp** [comment]
We should add new contributor guidelines, and that's an entirely reasonable thing to open an issue for. 

Depending on your level of comfort with TF-Java, there are plenty of things to work on. A fairly easy first step is to add more example models here - https://github.com/tensorflow/java-models, that will let you get comfortable with the API we have and help explain it to new users. For the library itself then we're working on building out a higher level framework API at the moment. Once the initial implementation of a dense layer has landed we'll need help to expand the set of layers to cover the same things Keras covers, which will involve building the layers themselves and also the lower level operations we need. Other things that would currently be helpful is trying to load in models that use tensorflow-text, tensorflow-io and the other TF modules that don't live in TF core. Our coverage of those modules is currently quite poor, so we're missing things like easy support for loading from s3 buckets e.g. #429. If you can identify things that don't work in those modules then we can help you figure out how to patch it in TF-Java and you can contribute those changes.

To supply any code to TF-Java you'll need to have signed a contributor license agreement with Google, which you can find more information about in the main TF contributor docs.

**Comment by GausiyaNisar** [comment]
For new contributors to TensorFlow, here are some steps and resources to get you started:

Read the CONTRIBUTING.mdfile: This file in the TensorFlow repository provides essential guidelines for contributing. You can find it here.

Sign the Contributor License Agreement (CLA): Before contributing code, you'll need to sign the CLA. This is a standard procedure for many open-source projects2.

Find a good first issue: TensorFlow tags issues with labels like "good first issue" and "contributions welcome" to help new contributors. These are great starting points to get familiar with the contribution workflow2.

Write code and tests: When you're ready to contribute, write your code and ensure it includes tests. This helps maintain the quality and reliability of the project3.

Submit a pull request: Once your code and tests are ready, submit a pull request (PR) on GitHub. Your PR will be reviewed by the TensorFlow team, and they may ask for changes or improvements2.

Participate in code reviews: Reviewing code submitted by other developers is also a valuable contribution. It helps improve the overall quality of the project and familiarizes you with the codebase2.

Join the TensorFlow community: Engage with the TensorFlow community through forums, mailing lists, and other communication channels. This can provide support, feedback, and collaboration opportunities

**Comment by Craigacp** [comment]
@GausiyaNisar Please don't submit LLM generated comments on issues.
